src/sentry/static/sentry/app/views/settings/projectDataPrivacy/projectDataPrivacyRulesForm.tsx

```diff
@@ -0,0 +1,148 @@
+import React from 'react';
+
+import styled from '@emotion/styled';
+import {t} from 'app/locale';
+import SelectControl from 'app/components/forms/selectControl';
+import TextField from 'app/components/forms/textField';
+import {IconDelete} from 'app/icons/iconDelete';
+import Button from 'app/components/button';
+
+enum ACTION_TYPE {
+  MASK = 'mask',
+  REMOVE = 'remove',
+  HASH = 'hash',
+  REPLACE = 'replace',
+}
+
+enum DATA_TYPE {
+  CREDIT_CARD_NUMBERS = 'Credit Card Numbers',
+  BANK_ACCOUNTS = 'Bank Accounts',
+  PASSWORD = 'Passwords',
+  PHONE_NUMBERS = 'Phone Numbers',
+  IP_ADDRESSES = 'IP Addresses',
+  CUSTOM_REGULAR_EXPRESSION = 'Custom Regular Expression',
+}
+
+export type RuleType = {
+  id: number;
+  action: ACTION_TYPE;
+  data: DATA_TYPE;
+  from: string;
+};
+
+type Props = {
+  onDelete?: () => void;
+} & RuleType;
+
+type State = Omit<RuleType, 'id'>;
+
+class ProjectDataPrivacyRulesForm extends React.Component<Partial<Props>, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      action: props.action,
+      data: props.data,
+      from: props.from,
+    };
+  }
+
+  handleChange = <T extends keyof State>(stateProperty: T, value: State[T]) => {
+    this.setState(prevState => ({
+      ...prevState,
+      [stateProperty]: value,
+    }));
+  };
+
+  handleOnDelete = (event: React.MouseEvent) => {
+    event.stopPropagation();
+    const {onDelete} = this.props;
+    onDelete && onDelete();
+  };
+
+  render() {
+    const {onDelete} = this.props;
+    return (
+      <Wrapper>
+        <StyledSelectControl
+          placeholder={t('Select an action')}
+          name="action"
+          options={Object.keys(ACTION_TYPE).map(key => ({
+            label: ACTION_TYPE[key],
+            value: key,
+          }))}
+          value={this.state.action}
+          onChange={({value}: {value: ACTION_TYPE}) => this.handleChange('action', value)}
+          maxWidth="157px"
+          openOnFocus
+        />
+
+        <StyledSelectControl
+          placeholder={t('Select data')}
+          name="data"
+          options={Object.keys(DATA_TYPE).map(key => ({
+            label: DATA_TYPE[key],
+            value: key,
+          }))}
+          value={this.state.data}
+          onChange={({value}: {value: DATA_TYPE}) => this.handleChange('data', value)}
+          maxWidth="300px"
```

Comment:
Do we need to think about smaller screen widths?

Reply:
sure ...I always leave it to the end ...